test/SlowTests/Issues/RavenDB_11632.cs

```diff
@@ -0,0 +1,98 @@
+using FastTests;
+using Xunit;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_11632 : RavenTestBase
+    {
+        [Fact]
+        public void ShouldBeAbleToStoreExistingDocumentWithEarlierIdAfterOldDocumentWasDeleted()
+        {
+            const string id = "docs/1";
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Document
+                    {
+                        Id = id,
+                        Text = "text-1"
+                    });
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var doc = session.Load<Document>(id);
+                    Assert.NotNull(doc);
+
+                    session.Delete(doc);
+
+                    doc = new Document
+                    {
+                        Id = id,
+                        Text = "text-2"
+                    };
+                    
+                    session.Store(doc);
+
+                    session.SaveChanges();
+                }
+                
+                using (var session = store.OpenSession())
+                {
+                    var doc = session.Load<Document>(id);
+                    Assert.NotNull(doc);
+                    Assert.Equal("text-2", doc.Text);
+                }
+            }
+        }
+        
+        [Fact]
+        public void ShouldBeAbleToStoreNewDocumentWithEarlierIdAfterOldDocumentWasDeleted()
+        {
+            const string id = "docs/1";
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Document
+                    {
+                        Id = id,
+                        Text = "text-1"
+                    });
+                    
+                    var doc = session.Load<Document>(id);
```

Comment:
This load it from the local session, mind

Reply:
That actually was the intent, to make sure local session returns the state that will be deleted, maybe a bit silly.